sphinxcontrib/datatemplates/directive.py

```diff
@@ -181,12 +196,57 @@ def run(self):
                 "-", "_")  # make identifier-compatible if trivially possible
             loader_options.setdefault(k, v)  # do not overwrite
 
-        with self.loader(**loader_options) as data:
-            context = self._make_context(data, app.config, env)
-            rendered_template = render_function(
-                template,
-                context,
-            )
+        try:
+            with self.loader(**loader_options) as data:
+                context = self._make_context(data, app.config, env)
+                rendered_template = render_function(
+                    template,
+                    context,
+                )
+        except FileNotFoundError:
+            error = self.state_machine.reporter.error(
+                f"Source file '{relative_resolved_path}' not found",
+                nodes.literal_block(self.block_text, self.block_text),
+                line=self.lineno)
+            return [error]
+        except ModuleNotFoundError:
+            error = self.state_machine.reporter.error(
+                f"Source module '{relative_resolved_path}' not found",
+                nodes.literal_block(self.block_text, self.block_text),
+                line=self.lineno)
+            return [error]
+        except (
+            json.decoder.JSONDecodeError,
+            ET.ParseError,
+            dbm.error[0],
+        ) as err:
+            error = self.state_machine.reporter.error(
+                f"Error in source file '{relative_resolved_path}': {err}",
+                nodes.literal_block(self.block_text, self.block_text),
+                line=self.lineno)
+            return [error]
+        except yaml.error.MarkedYAMLError as err:
```

Comment:
We should make it the responsibility of the loader to convert these format-specific exceptions (YAML, JSON, etc.) into a consistent type defined in the datatemplates package. That way, if we add new data formats and sources in the future we do not have to extend this exception handling block further.

The file and template error handling can stay as it is, since those apply regardless of the source or type of data.

Reply:
Ok, done.